workspaces/homepage/plugins/dynamic-home-page/src/components/EntitySection/EntitySection.tsx

```diff
@@ -0,0 +1,268 @@
+        <CodeSnippet
+          language="text"
+          text={error?.toString() ?? 'Unknown error'}
+        />
+      </WarningPanel>
+    );
+  } else {
+    content = (
+      <Box sx={{ padding: '20px 10px 10px 0' }}>
+        <React.Fragment>
+          <Grid container spacing={1} alignItems="stretch">
+            {!isRemoveFirstCard && (
+              <Grid item xs={12} md={5} key="entities illustration">
+                <Box
+                  sx={{
+                    border: `1px solid ${theme.palette.grey[400]}`,
+                    borderRadius: 3,
+                    display: 'flex',
+                    alignItems: 'center',
+                    position: 'relative',
+                    transition:
+                      'opacity 0.5s ease-out, transform 0.5s ease-in-out',
+                    opacity: showDiscoveryCard ? 1 : 0,
+                    transform: showDiscoveryCard
+                      ? 'translateX(0)'
+                      : 'translateX(-50px)',
+                  }}
+                >
+                  {!imgLoaded && (
+                    <Skeleton
+                      variant="rectangular"
+                      width={imageWidth}
+                      height={300}
+                      sx={{ borderRadius: 3 }}
+                    />
+                  )}
+                  <Box
+                    component="img"
+                    src={HomePageEntityIllustration}
+                    onLoad={() => setImgLoaded(true)}
+                    alt="entities illustration"
+                    height={300}
+                    width={imageWidth}
+                  />
+                  <Box sx={{ p: 2 }}>
+                    <Box>
+                      <Typography variant="body2" paragraph>
+                        Browse the Systems, Components, Resources, and APIs that
+                        are available in your organization.
+                      </Typography>
+                    </Box>
+                    {entities?.length > 0 && (
+                      <IconButton
+                        onClick={handleClose}
+                        aria-label="close"
+                        style={{
+                          position: 'absolute',
+                          top: '8px',
+                          right: '8px',
+                        }}
+                      >
+                        <CloseIcon style={{ width: '16px', height: '16px' }} />
+                      </IconButton>
+                    )}
+                  </Box>
+                </Box>
+              </Grid>
+            )}
+            {entities?.slice(0, isRemoveFirstCard ? 4 : 2).map((item: any) => (
+              <Grid
+                item
+                xs={12}
+                md={isRemoveFirstCard ? 3 : 3.5}
+                key={item.metadata.name}
+              >
+                <CardWrapper
+                  link={`/catalog/default/resource/${item.metadata.name}`}
```

Comment:
`kind` is hardcoded and it redirects you to 404 page.

```suggestion
                  link={`/catalog/default/${item.kind}/${item.metadata.name}`}
```